rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1521,6 +1514,50 @@ public Coin getMinimumLockTxValue() {
         return bridgeConstants.getMinimumLockTxValue();
     }
 
+    /**
+     * Votes for a fee per kb value.
+     *
+     * @return 2 upon successful vote with election winner,
+     * 1 upon successful vote without winner,
+     * -1 when the vote failed,
+     * FEE_PER_KB_GENERIC_ERROR_CODE when there was an un expected error.
+     */
+    public Integer voteFeePerKbChange(Transaction tx, Coin feePerKb) {
+        TxSender voter = TxSender.fromTx(tx);
+        ABICallElection feePerKbElection = provider.getFeePerKbElection(bridgeConstants.getFeePerKbChangeAuthorizer());
+        ABICallSpec feeVote = new ABICallSpec("setFeePerKb", new byte[][]{BridgeSerializationUtils.serializeCoin(feePerKb)});
+        boolean successfulVote = feePerKbElection.vote(feeVote, voter);
+        if (!successfulVote) {
+            return -1;
+        }
+
+        ABICallSpec winner = feePerKbElection.getWinner();
+        if (winner == null) {
+            return 1;
+        }
+
+        Coin winnerFee;
+        try {
+            winnerFee = BridgeSerializationUtils.deserializeCoin(winner.getArguments()[0], null);
+        } catch (Exception e) {
+            logger.warn("Exception deserializing winner feePerKb", e);
+            return FEE_PER_KB_GENERIC_ERROR_CODE;
+        }
+
+        if (winnerFee == null) {
+            logger.warn("Invalid winner feePerKb: feePerKb can't be null");
+            return FEE_PER_KB_GENERIC_ERROR_CODE;
+        }
+
+        if (!winnerFee.equals(feePerKb)) {
+            logger.debug("Winner fee is different than the last vote: maybe you forgot to clear winners");
+        }
+
+        provider.setFeePerKb(winnerFee);
+        feePerKbElection.clearWinners();
+        return 2;
```

Comment:
We're not differentiating between successful vote with and without winner in the federation election. We probably don't want to do it here either.